app/data/app/cache-strategy/IAppCacheDataSource.ts

```diff
@@ -0,0 +1,7 @@
+export default interface IAppCacheDataSource {
```

Comment:
Since this data source is related to app preferences, the full file path can be `app/data/app/IAppDataSource` and its implementation as `app/local/app/AppPreferences`.